### ISSUE TITLE
K3S secret-encryption feature enhancement to support  - decrypt all secrets, Rotating a decryption key

### ISSUE THREAD

**Issue by vakrat1** [k3s-io/k3s#3617]
Hi
We'd like to use the `--secrets-encryption` option to decrypt the Secrets we create on our K3S deployed cluster.
According to the documentation, the K3s impl will do the following:

> - Generate an AES-CBC key
> - Generate an encryption config file with the generated key
> - Pass the config to the KubeAPI as encryption-provider-config
> 
> Once enabled any created secret will be encrypted with this key

We have a use case, that while Upgrading our system we might also upgrade the K3S version/deployment.
What would happen to the `AES-CBC key` in that case ? Would it be regenerated ?  If that so, would it mean that we need to decrypt all our Secrets first. 

Is it possible to disable the `--secrets-encryption` and rely on K3S to decrypt all the Secrets resources?

Another issue is the option to **Rotating a decryption key** - is it possible to rotate the `AES-CBC key` ? 
We need it in order for being compatible with security regulations.

Bottom line, we need to know if we could have access to the ` encryption config file` that is being generated by the K3S to manipulate its configuration (both the key and the order of the providers)

Tnx

**Comment by brandond** [comment]
This is being tracked under https://github.com/k3s-io/k3s/issues/3407
